From c088688bdec650b98c1534163875f25fd1d4dd2d Mon Sep 17 00:00:00 2001
From: Robert de Bath <rdebath@tvisiontech.co.uk>
Date: Sat, 7 Nov 2015 22:02:57 +0000
Subject: [PATCH 10/71] ITU-T T.416 truecolour (fuzzterm)

---
 fuzzterm.c | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/fuzzterm.c b/fuzzterm.c
index 15b5d63..b5193ec 100644
--- a/fuzzterm.c
+++ b/fuzzterm.c
@@ -45,22 +45,22 @@ int from_backend(void *frontend, int is_stderr, const char *data, int len)
 
 void request_resize(void *frontend, int x, int y) { }
 void do_text(Context ctx, int x, int y, wchar_t * text, int len,
-	     unsigned long attr, int lattr)
+	     unsigned long attr, int lattr, int fg_col, int bg_col)
 {
     int i;
 
-    printf("TEXT[attr=%08lx,lattr=%02x]@(%d,%d):", attr, lattr, x, y);
+    printf("TEXT[attr=%08lx,lattr=%02x,fg=%d,bg=%d]@(%d,%d):", attr, lattr, fg_col, bg_col, x, y);
     for (i = 0; i < len; i++) {
 	printf(" %x", (unsigned)text[i]);
     }
     printf("\n");
 }
 void do_cursor(Context ctx, int x, int y, wchar_t * text, int len,
-	     unsigned long attr, int lattr)
+	     unsigned long attr, int lattr, int fg_col, int bg_col)
 {
     int i;
 
-    printf("CURS[attr=%08lx,lattr=%02x]@(%d,%d):", attr, lattr, x, y);
+    printf("CURS[attr=%08lx,lattr=%02x,fg=%d,bg=%d]@(%d,%d):", attr, lattr, fg_col, bg_col, x, y);
     for (i = 0; i < len; i++) {
 	printf(" %x", (unsigned)text[i]);
     }
-- 
2.1.4

